crates/sequencer/src/serde/state.rs

```diff
@@ -0,0 +1,156 @@
+use std::{fs, path::PathBuf};
+
+use blockifier::state::cached_state::ContractStorageKey;
+use rustc_hash::FxHashMap;
+use serde::{Deserialize, Serialize};
+use starknet_api::{
+    core::{ClassHash, CompiledClassHash, ContractAddress, Nonce},
+    hash::StarkFelt,
+};
+use thiserror::Error;
+
+use crate::state::State;
+
+use super::contract::SerializableContractClass;
+
+/// TODO: wrap the underlying errors in them
+#[derive(Error, Debug)]
+pub enum SerializationError {
+    #[error("{reason:?}")]
+    IoError {
+        reason: String,
+        context: std::io::Error,
+    },
+    #[error("{reason:?}")]
+    SerdeJsonError {
+        reason: String,
+        context: serde_json::Error,
+    },
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, Default)]
+pub struct SerializableState {
+    pub classes: FxHashMap<ClassHash, SerializableClassRecord>,
+    pub compiled_classes_hash: FxHashMap<ClassHash, CompiledClassHash>,
+    pub contracts: FxHashMap<ContractAddress, ClassHash>,
+    #[serde(with = "serialize_contract_storage")]
+    pub storage: FxHashMap<ContractStorageKey, StarkFelt>,
+    pub nonces: FxHashMap<ContractAddress, Nonce>,
+}
+
+impl SerializableState {
+    pub fn dump_state(&self) -> Result<Vec<u8>, SerializationError> {
+        let serialized_state_buffer =
+            serde_json::to_vec(&self).map_err(|error| SerializationError::SerdeJsonError {
+                reason: "Failed at serializing state to a buffer".to_string(),
+                context: error,
+            })?;
+
+        Ok(serialized_state_buffer)
+    }
+
+    pub fn load_state(path: &PathBuf) -> Result<Self, SerializationError> {
```

Comment:
could we change this to `new`? `new` could take an Option<PathBuf> and load the file if Option is not None.

Reply:
I think, `from_file_path` is a better approach, it should be a static method and do one single thing!

We can see this pattern in other rust codebases as well, where there is options like `from_string`, etc.

Reply:
Lemme know what you think about it.